packages/common/base/src/blockchain/utils/isBlockchainIncludingTestnets.ts

```diff
@@ -0,0 +1,10 @@
+import { BLOCKCHAINS_INCLUDING_TESTNETS, Blockchain, BlockchainIncludingTestnet } from "../types";
+
+export function isBlockchainIncludingTestnets<T extends BlockchainIncludingTestnet = BlockchainIncludingTestnet>(
+    value: unknown,
+    expectedBlockchain?: T
+): value is T {
+    return expectedBlockchain
+        ? value === expectedBlockchain
```

Comment:
how is this function used? we have logic that switches on a param to decide if the entire function amounts to `value === expectedBlockchain`? 
feels wack

Reply:
this is taken from crossmint-main

its pretty good imo, bc it allows you to build every single `isEVM, isSolana, isSOMECHAIN` all from this single function

you can see how its used in `isEVMBlockchain`

Reply:
which i think is better than this

from aa sdk
![image](https://github.com/Crossmint/crossmint-sdk/assets/93682696/1a3e7b97-083d-41a5-9d1d-fceab03d1c20)